File: fizzbuzz/fizzbuzz.exs
defmodule FizzBuzz do
  def fizz(n), do: check(n, 3, "Fizz")
  def buzz(n), do: check(n, 5, "Buzz")

  defp check(number, value, output) do
    if rem(number, value) == 0 do
      output
    else
      ""
    end
  end

  def run(range) do
    range
    |> Enum.map(&max(fizz(&1) <> buzz(&1), "#{&1}"))
    |> Enum.map(&IO.puts/1)
  end
end

FizzBuzz.run(1..100)

File: fizzbuzz/fizzbuzz.hs
module Main where

fizzbuzz :: Int -> String
fizzbuzz n = max (show n) (fizz n ++ buzz n)
  where
    check :: Int -> String -> Int -> String
    check v out n =
      if n `rem` v == 0
        then out
        else ""
    fizz = check 3 "Fizz"
    buzz = check 5 "Fuzz"

main = mapM_ putStrLn $ map fizzbuzz [1 .. 100]
